webtau-http/src/main/java/org/testingisdocumenting/webtau/http/datanode/DataNodeCompareToHandler.java

```diff
@@ -55,10 +56,10 @@ public void compareGreaterLessEqual(CompareToComparator compareToComparator, Act
         compareToComparator.compareUsingCompareTo(actualPath, actualDataNode.getTraceableValue(), expected);
     }
 
-    private void compareWithMap(CompareToComparator comparator, ActualPath actualPath, DataNode actual, Map expected) {
-        Map<String, DataNode> actualAsMap = actual.asMap();
+    private void compareWithMap(CompareToComparator comparator, ActualPath actualPath, DataNode actual, Map<?, ?> expected) {
+        Map<String, DataNode> actualAsMap = DataNodeToMapConverter.convertToMap(actual);
```

Comment:
I think instead of converting to a map here, we should do this:
```java
    private void compareWithMap(CompareToComparator comparator, ActualPath actualPath, DataNode actual, Map expected) {
        Set keys = expected.keySet();
        for (Object key : keys) {
            String p = (String) key;
            ActualPath propertyPath = actualPath.property(p);

            Object expectedValue = expected.get(p);
            if (! actual.has(p)) {
                comparator.reportMissing(this, propertyPath, expectedValue);
            } else {
                comparator.compareUsingEqualOnly(propertyPath, actual.get(p), expectedValue);
            }
        }
    }
```
Then all operations are directly on the node therefore any future null tracking (😁 ) will be aware of these access attempts.

Reply:
good idea